sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -89,7 +89,8 @@ az_write_to_buffer(az_span writable_buffer, az_pair header, az_span separator) {
   return AZ_OK;
```

Comment:
Is the line above the bug? You probably don't append a zero-byte; don't you append a CRLF

Reply:
Ah! And here's possibly why it now fails when I add Content-Type header for the auth!